code/game/objects/items/stacks/money.dm

```diff
@@ -212,6 +239,40 @@
 		value = 4
 		return ..()
 
+/obj/item/stack/money/dollar100
+	name = "100 Dollar Bill"
+	desc = "Paper bank note valued at one-hundred dollars"
+	icon_state = "100dollar"
+	value = 100
+	amount = 1
+
+/obj/item/stack/money/dollar100/New()
+	if(amount < 50)
+		icon_state = "100dollar"
+	if(amount >= 50)
+		icon_state = "100dollar50"
+	if(amount >= 100)
+		icon_state = "100dollar100"
+	if(amount >= 300)
+		icon_state = "100dollar300"
+	if(amount == 500)
+		icon_state = "100dollar500"
+	update_icon()
+	return ..()
+
+/obj/item/stack/money/dollar100/update_icon()
+	if(amount < 50)
+		icon_state = "100dollar"
+	if(amount >= 50)
+		icon_state = "100dollar50"
+	if(amount >= 100)
+		icon_state = "100dollar100"
+	if(amount >= 300)
+		icon_state = "100dollar300"
+	if(amount == 500)
+		icon_state = "100dollar500"
+	..()
```

Comment:
However we don't need to call the parent proc's `icon_state()` here as it does nothing of use to us and could even potentially involve bugs, and slow the code down:
`
/obj/item/stack/money/update_icon()
	if(novariants)
		return ..()
	if(amount >= 2)
		if (map.ordinal_age >= 4)
			if(icon_state != "silvercoin_pile")
				icon_state = "[initial(icon_state)]_2"

	else
		if (map.ordinal_age >= 4)
			icon_state = initial(icon_state)
	..()`